src/protocol/sort/generate_sort_permutations.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    error::BoxError,
+    ff::Field,
+    protocol::{
+        context::ProtocolContext,
+        modulus_conversion::convert_shares::convert_all_shares,
+        sort::bit_permutations::BitPermutations,
+        sort::SortStep::{ApplyInv, BitPermutation, ComposeStep},
+    },
+    secret_sharing::Replicated,
+};
+
+use super::{compose::Compose, secureapplyinv::SecureApplyInv};
+use embed_doc_image::embed_doc_image;
+
+/// This is an implementation of `GenPerm` (Algorithm 6) described in:
+/// "An Efficient Secure Three-Party Sorting Protocol with an Honest Majority"
+/// by K. Chida, K. Hamada, D. Ikarashi, R. Kikuchi, N. Kiribuchi, and B. Pinkas
+/// <https://eprint.iacr.org/2019/695.pdf>.
+#[derive(Debug)]
+pub struct GenerateSortPermutations<'a> {
```

Comment:
Why’s it plural? It just generates a single permutation does it not?

Reply:
True.. Let me change it